appserver/batch/glassfish-batch-connector/src/main/java/fish/payara/jbatch/persistence/rdbms/MySqlPersistenceManager.java

```diff
@@ -272,11 +272,12 @@ protected void createMySQLTableNotExists(String tableName,
     }
 
     @Override
-    protected void setSchemaOnConnection(Connection connection) throws SQLException {
-            PreparedStatement ps = null;
-            ps = connection.prepareStatement("USE " + schema);
-            ps.executeUpdate();
-            ps.close();
+    protected void setSchemaOnConnection(Connection connection) throws SQLException { 
+        logger.log(Level.FINEST, "Entering {0}.setSchemaOnConnection()", CLASSNAME);
+        try (PreparedStatement preparedStatement = connection.prepareStatement("USE " + schema)) {
+            preparedStatement.executeUpdate();
+        }
+        logger.log(Level.FINEST, "Exiting {0}.setSchemaOnConnection()", CLASSNAME);
```

Comment:
@MeroRai This log will not be called in case of an SQLException. You might want to consider putting this in a finally block.